cmake/kokkos_arch.cmake

```diff
@@ -375,6 +376,18 @@ IF (KOKKOS_ARCH_HSW)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_RISCV_MILKV)
+  COMPILER_SPECIFIC_FLAGS(
+    COMPILER_ID KOKKOS_CXX_HOST_COMPILER_ID
+        Cray    NO-VALUE-SPECIFIED
+        Intel   NO-VALUE-SPECIFIED
+        MSVC    NO-VALUE-SPECIFIED
+        NVHPC   NO-VALUE-SPECIFIED
+        DEFAULT -march=rv64imafdcv
```

Comment:
Are you able to give us more context?  What toolchain is it intended to work with?
I would prefer to error out at configuration time if the compiler does not know about RISCV rather than silently do nothing.  Otherwise will run into problems later when we actually have code that is guarded for that architecture.

Reply:
I only have GCC 12 and GCC 13 on my RISC-V machines and tested with both of them.

So far, I can only confirm that I could build kokkos with GCC 12 and GCC 13. 

I am doing some scaling runs and they do look good.

Reply:
it seems LLVM has RISCV support [now](https://riscv.org/news/2019/09/llvm-9-releases-with-official-risc-v-target-support-asm-goto-clang-9-and-more-vincy-davis-packt-pub/). But I have no chance to test that.

Reply:
Then do something like
```cmake
IF(NOT (
  (KOKKOS_CXX_COMPILER_ID STREQUAL GNU
    AND KOKKOS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12)
  OR
  (KOKKOS_CXX_COMPILER_ID STREQUAL Clang 
    AND KOKKOS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 16)
)
MESSAGE(SEND_ERROR "not supported because bla")
ENDIF()
```

Reply:
Ok, I added that.